gatsby/src/pages/search.tsx

```diff
@@ -1,14 +1,278 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { PageRendererProps, Link } from "gatsby";
+import parser from "react-html-parser";
+import { Card } from "@nice-digital/nds-card";
+import { SimplePagination } from "@nice-digital/nds-simple-pagination";
+import { Alert } from "@nice-digital/nds-alert";
+
 import { Layout } from "../components/Layout/Layout";
 import { SEO } from "../components/SEO/SEO";
 
-const SearchPage: React.FC = () => {
+type SearchPageProps = PageRendererProps;
+
+interface SearchResults {
+	failed: boolean;
+	resultCount: number;
+	firstResult: number;
+	lastResult: number;
+	finalSearchText: string;
+	finalSearchTextNoStopWords: string;
+	pageSize: number;
+	originalSearch: {
+		searchText: string;
+	};
+	pagerLinks: {
+		previous: {
+			fullUrl: string;
+		} | null;
+		next: {
+			fullUrl: string;
+		} | null;
+		pages: {
+			title: string;
+			url: {
+				fullUrl: string;
+			};
+			isCurrent: boolean;
+		}[];
+	};
+	documents: Document[];
+}
+
+type Document = {
+	id: string;
+	metaDescription: string;
+	contentId: string;
+	pathAndQuery: string;
+	sourceUrl: string;
+	teaser: string;
+	title: string;
+	url: string;
+};
+
+function titleString(text: string, page: number): string {
+	return text
+		? page > 1
+			? `Page ${page} | ${text} | `
+			: `${text} | `
+		: page > 1
+		? `Page ${page} | `
+		: "";
+}
+
+const SearchPage: React.FC<SearchPageProps> = ({
+	location,
+}: SearchPageProps) => {
+	const [data, setData] = useState<SearchResults | null>(null);
+	const [error, setError] = useState<Error>();
+
+	useEffect(() => {
+		setData(null);
+		fetch("/api/search" + window.location.search)
+			.then(data => data.json())
+			.then(results => setData(results as SearchResults))
+			.catch(e => setError(e));
+	}, [location.search]);
+
 	return (
 		<Layout>
-			<SEO title="Search" />
 			<h1>Search</h1>
+			{!error && !data && <div>Loading</div>}
+			{data && data.failed && <div>Failed to load results</div>}
+			{data && !data.failed && <Results {...data} />}
+			{error?.message && (
+				<Alert type="error">
+					<pre>
+						{error.name}
+						<br />
+						{error.message}
+						<br />
+						{error.stack}
+					</pre>
+				</Alert>
+			)}
 		</Layout>
 	);
 };
 
+const Results: React.FC<SearchResults> = ({
+	firstResult,
+	resultCount,
+	finalSearchText,
+	originalSearch,
+	documents,
+	pageSize,
+	pagerLinks: { next, previous },
+}) => {
+	const currentPage = Math.ceil(firstResult / pageSize);
+	const totalPages = Math.ceil(resultCount / pageSize);
+	const paginationProps = {
+		nextPageLink: next?.fullUrl
```

Comment:
Can we just use && here rather than a ternary?

Reply:
So this was a TS thing - it's upset unless you explicitly pass it undefined? 
<img width="2025" alt="Screenshot 2020-06-18 at 11 57 38" src="https://user-images.githubusercontent.com/6341474/85012953-bda3f280-b15b-11ea-85b4-954bf34d10eb.png">